src/core/src/core_logic/PatchInstaller.py

```diff
@@ -98,6 +106,16 @@ def start_installation(self, simulate=False):
 
         return overall_patch_installation_successful
 
+    def write_installer_perf_logs(self, patch_operation_successful, installed_patch_count, number_of_rounds, maintenance_window, maintenance_window_exceeded, task_status, error_msg):
+        perc_maintenance_window_used = maintenance_window.get_percentage_maintenance_window_used()
```

Comment:
IF get_percentage_maintenance_window_used() throws an exception, it goes all the way up and will stop the extension execution. i.e. a failure in computing an writing perf logs stops execution. 

Checking if that is okay? @kjohn-msft @GAURAVRAMRAKHYANI @SathishMSFT

Reply:
Question applies to all Stopwatch functions (i.e. start, stop, stop_and_write_telemetry) where an Exception is raised

Reply:
Raising exceptions in Stopwatch functions and get_percentage_maintenance_window_used on unexpected scenarios as they are common code that are going to be used in different places. 

Swallowing exception in write_installer_perf_logs and write_assessment_perf_logs as they are specific to writing logs.

@rane-rajasi @kjohn-msft @SathishMSFT